When nspawn fails immediately, it does not get logged

When starting a  job that exits immediately (e.g. due to wrong config in nginx) the driver think the job is failed because it doesn't show up in machinectl :

```
    2020-07-11T14:18:13.777+0200 [ERROR] client.driver_mgr.nomad-driver-nspawn: failed to get machine information: driver=nspawn @module=nspawn error="timed out while getting machine properties: No machine 'example-6e0636df-97ce-5a06-b354-ce79a7b7c63a' known" timestamp=2020-07-11T14:18:13.776+0200
```

because probably the service is already gone at that point; and will not show up in `machinectl`

Instead I would expect logs to show up when viewing the `alloc` in  nomad, but those are now empty because the allocation is never considered succesfully created

```
ID        Node ID   Task Group  Version  Desired  Status  Created    Modified
694b6326  430116cf  example     0        run      failed  2m10s ago  1m39s 
nomad alloc logs 694b6326
Error reading file: Unexpected response code: 404 (task "example" not started yet. No logs available)
```

This makes it very hard to debug to figure out why an `nspawn` command failed.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
   - Support for Nomad `bridge` networking mode. This also enables the use of
     Consul Connect.
+  - Output the error why systemd-nspawn fails on startup. It is added to the
+    logs and shown in the WebUi.
 ## [0.2.0] - 2020-06-27
 ### Added
   - Support for downloading images via the

--- a/nspawn/driver.go
+++ b/nspawn/driver.go
@@ -3,9 +3,11 @@ package nspawn
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strconv"
+	"strings"
 	"time"
 
 	"github.com/hashicorp/consul-template/signals"
@@ -433,16 +435,48 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 		Resources:  cfg.Resources,
 	}
 
-	_, err = exec.Launch(execCmd)
+	ps, err := exec.Launch(execCmd)
 	if err != nil {
 		pluginClient.Kill()
 		return nil, nil, fmt.Errorf("failed to launch command with executor: %v", err)
 	}
 
-	// wait for boot
+	printErr := func() {
+		logDir := cfg.TaskDir().LogDir
+		logs, err := filepath.Glob(filepath.Join(logDir, cfg.Name+"*"))
+		if err != nil {
+			d.logger.Error("error finding log files", err)
+			return
+		}
+
+		for _, l := range logs {
+			out, err := ioutil.ReadFile(l)
+			if err != nil {
+				continue
+			}
+			lines := strings.Split(strings.Trim(string(out), "\n"), "\n")
+			// Continue if there's no output
+			if len(lines) == 0 || len(lines[len(lines)-1]) == 0 {
+				continue
+			}
+			d.logger.Error("systemd-nspawn failed", "file", filepath.Base(l), "out", lines[len(lines)-1])
+			d.eventer.EmitEvent(&drivers.TaskEvent{
+				TaskID:    cfg.ID,
+				AllocID:   cfg.AllocID,
+				TaskName:  cfg.Name,
+				Timestamp: time.Now(),
+				Message:   lines[len(lines)-1],
+				Err:       fmt.Errorf("Systemd-Nspawn failed"),
+			})
+		}
+	}
+
 	p, err := DescribeMachine(driverConfig.Machine, machinePropertiesTimeout)
 	if err != nil {
 		d.logger.Error("failed to get machine information", "error", err)
+		if ps.ExitCode != 0 {
+			printErr()
+		}
 		if !pluginClient.Exited() {
 			if err := exec.Shutdown("", 0); err != nil {
 				d.logger.Error("destroying executor failed", "err", err)
@@ -457,6 +491,9 @@ func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *drive
 	addr, err := MachineAddresses(driverConfig.Machine, machineAddressTimeout)
 	if err != nil {
 		d.logger.Error("failed to get machine addresses", "error", err, "addresses", addr)
+		if ps.ExitCode != 0 {
+			printErr()
+		}
 		if !pluginClient.Exited() {
 			if err := exec.Shutdown("", 0); err != nil {
 				d.logger.Error("destroying executor failed", "err", err)

